### PR TITLE
refactor: seprate ext js file and organize them

### PR DESCRIPTION
**Pull request by LiKang6688**
Did not change any logic code, just separate a file.

### CODE REVIEW

File: src/extension-metadata/__test__/data.spec.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
+import { indexAdded, indexRemoved, min, getDescription } from '../data';
```
Comment (haxxmaxx) [inline src/extension-metadata/__test__/data.spec.js:1]: Is all of this code new?

**Review by haxxmaxx — COMMENTED** [review]
This is a good idea, but not sure if it is a great time to do it. It was not planned and there is the new definition for the styling pane. We might as well wait for that before we start refactoring this

File: src/extension-metadata/__test__/data.spec.js (hunk 1 above)
Comment (LiKang6688) [reply src/extension-metadata/__test__/data.spec.js:1]: No, they are not.

Comment (LiKang6688) [reply src/extension-metadata/__test__/data.spec.js:1]: I mean this line of code is new. But those methods are not.

**Comment by LiKang6688** [comment]
> This is a good idea, but not sure if it is a great time to do it. It was not planned and there is the new definition for the styling pane. We might as well wait for that before we start refactoring this

I agree with you. Let's wait.

File: src/extension-metadata/data.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+export function min(nDimsOrMeas) {
+  return nDimsOrMeas > 0 ? 0 : 1;
+}
+
+export function getDescription(env) {
+  return env.translator.get('Visualizations.Descriptions.Column');
+}
+
+export default function setData(env) {
```
[the remark this answers is not in the document]
Comment (LiKang6688) [reply src/extension-metadata/data.js]: fixed

File: src/index.ts [hunk 3]
```diff
@@ -18,7 +18,7 @@ import {
 
 import properties from './object-properties';
 import data from './data';
-import ext from './ext';
+import ext from './extension-metadata';
```
[the remark this answers is not in the document]
Comment (LiKang6688) [reply src/index.ts]: fixed

**Comment by haxxmaxx** [comment]
Closing the PR for now. Feel free to reopen when this is being worked on

**Pull request closed without merging**